clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -909,6 +909,39 @@ public void ensureValid(String name, Object o) {
         }
     }
 
+    public static class ValidName implements Validator {
+        final char[] illegalCharacters = {'/', ' ', '\\', '\n'};
+
+        public static ValidName validName() {
+            return new ValidName();
+        }
+
+        @Override
+        public void ensureValid(String name, Object value) {
+            String s = (String) value;
+
+            if (s == null) {
+                // This can happen during creation of the config object due to no default value being defined for the
+                // name configuration - a missing name parameter is caught when checking for mandatory parameters,
+                // thus we can ok a null value here
+                return;
+            } else if (s.isEmpty()) {
+                throw new ConfigException(name, value, "Name may not be empty");
+            }
+
+            // Check name string for illegal characters
```

Comment:
Hi Viktor, I considered it, yes, but decided on this way because it saves us from building a regex using string operations, which to my mind always bears the potential of escaping issues. 
The alternative would be to directly have the regexp in the code, which tbh I find considerably less readable than the way it currently is.
I figured that performance is less of an issue for this, since adding a connector will not be called that often, but to be on the safe side I just did some quick'n dirty tests on performance and a precompiled regex was actually a bit slower than the indexOf way. But I will grant that there may be some tuning potential in the regex code there.